cmd/argo/commands/wait.go

```diff
@@ -91,8 +91,12 @@ func waitOnOne(serviceClient workflowpkg.WorkflowServiceClient, ctx context.Cont
 			log.Debug("Re-establishing workflow watch")
 			stream, err = serviceClient.WatchWorkflows(ctx, req)
 			errors.CheckError(err)
+			continue
 		}
 		errors.CheckError(err)
+		if event == nil {
+			continue
```

Comment:
why is event nil?

Reply:
This is an extra check based on stack trace event is nil for panic.  Based on code, if re-establishment happens, an event may be nil.